.github/workflows/ci.yml

```diff
@@ -0,0 +1,40 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - master
+      - '*'
+
+jobs:
+  scalar-ist-ci-tests:
+
+    runs-on: ubuntu-18.04
+    env:
+      CONTRACTS_AND_FUNCTIONS_DIR: ${{ github.workspace }}/contracts_and_functions
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Cache
+        uses: actions/cache@v1.1.2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Run unit tests
+        run: |
+          cd ${{ env.CONTRACTS_AND_FUNCTIONS_DIR }}
+          ./gradlew test
```

Comment:
Doesn't test also run build?

Reply:
@ypeckstadt san,
Yes, you are correct.
Sorry, it's my mistake.

I think we can run unit tests and build creation in separate steps for easy understanding.
Can I modify it like this?
```
      - name: Run unit tests
        run: |
          cd ${{ env.CONTRACTS_AND_FUNCTIONS_DIR }}
          ./gradlew test

      - name: Build scalar-ist contracts and functions
        run: |
          cd ${{ env.CONTRACTS_AND_FUNCTIONS_DIR }}
          ./gradlew build -x test
```

Or do I need to remove the step `Run unit tests`?

Reply:
I checked the other existing actions we have for other projects, for example, dataloader. And there we only run the following:

```
- name: Gradle check
        run: ./gradlew check
```

Reply:
Thank you for your suggestion.
PR revised based on review comments.